src/query_plan/operation.rs

```diff
@@ -873,6 +943,41 @@ pub(crate) mod normalized_fragment_spread_selection {
         pub(crate) fn data(&self) -> &NormalizedFragmentSpreadData {
             &self.data
         }
+
+        pub(crate) fn can_add_to(
+            &self,
+            parent_type: &CompositeTypeDefinitionPosition,
+            schema: &ValidFederationSchema,
+        ) -> Result<bool, FederationError> {
```

Comment:
Remaining question: I'm not sure how to port this from JS. Inheritance made this hard to parse out. A pointer on where to look would be helpful.